File: src/mastodon/types/status.ts
/**
 * Mastodon Status Types
 * @see https://docs.joinmastodon.org/entities/Status/
 */

import { Account, Emoji } from './account';
import { MediaAttachment } from './media';

export interface Status {
  id: string;
  created_at: string;
  in_reply_to_id: string | null;
  in_reply_to_account_id: string | null;
  sensitive: boolean;
  spoiler_text: string;
  visibility: StatusVisibility;
  uri: string;
  url: string;
  replies_count: number;
  reblogs_count: number;
  favourites_count: number;
  favourited: boolean;
  reblogged: boolean;
  muted: boolean;
  bookmarked: boolean;
  pinned: boolean;
  content: string;
  reblog: Status | null;
  application: Application | null;
  account: Account;
  media_attachments: MediaAttachment[];
  mentions: Mention[];
  tags: Tag[];
  emojis: Emoji[];
  card: Card | null;
  poll: Poll | null;
  reply_to?: Status;
  ancestors?: Status[];
  thread_continuation?: Status[];
}

export type StatusVisibility = 'public' | 'unlisted' | 'private' | 'direct';

export interface Application {
  name: string;
  website: string | null;
}

export interface Mention {
  id: string;
  username: string;
  url: string;
  acct: string;
}

export interface Tag {
  name: string;
  url: string;
}

export interface Card {
  url: string;
  title: string;
  description: string;
  type: 'link' | 'photo' | 'video' | 'rich';
  author_name: string;
  author_url: string;
  provider_name: string;
  provider_url: string;
  html: string;
  width: number;
  height: number;
  image: string;
}

export interface Poll {
  id: string;
  expires_at: string;
  expired: boolean;
  multiple: boolean;
  votes_count: number;
  voters_count?: number;
  options: PollOption[];
  emojis: Emoji[];
  voted?: boolean;
  own_votes?: number[];
}

export interface PollOption {
  title: string;
  votes_count: number;
}

export interface StatusContext {
  ancestors: Status[];
  descendants: Status[];
}

export interface ScheduledStatusParams {
  text?: string;
  warning_text?: string;
  visibility?: StatusVisibility;
  sensitive?: string;
  language?: string;
  media_ids?: string[];
  scheduled_at?: string;
  poll?: {
    options?: string[];
    expires_in?: string;
    multiple?: string;
  };
}

export interface ScheduledStatus {
  id: string;
  scheduled_at: string;
  params?: ScheduledStatusParams;
  media_attachments?: MediaAttachment[];
}

/**
 * @deprecated Use Status instead - this alias exists for backwards compatibility
 */
export type Post = Status;

File: android/wear/src/main/java/place/coho/app/wear/api/models/Status.kt
package place.coho.app.wear.api.models

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = false)
data class Status(
    val id: String,
    @Json(name = "created_at") val createdAt: String,
    @Json(name = "in_reply_to_id") val inReplyToId: String?,
    val sensitive: Boolean = false,
    @Json(name = "spoiler_text") val spoilerText: String = "",
    val visibility: String = "public",
    val content: String, // HTML content
    val reblog: Status?,
    val account: Account,
    @Json(name = "media_attachments") val mediaAttachments: List<MediaAttachment> = emptyList(),
    @Json(name = "replies_count") val repliesCount: Int = 0,
    @Json(name = "reblogs_count") val reblogsCount: Int = 0,
    @Json(name = "favourites_count") val favouritesCount: Int = 0,
    val favourited: Boolean = false,
    val reblogged: Boolean = false,
)
